tests/base_scraper/test_base_scraper.py

```diff
@@ -74,5 +82,28 @@ def test_notimplementedexceptions(self):
         self.assertRaises(mozdownload.NotImplementedError,
                           scraper.build_filename, 'invalid binary')
 
+    def test_destination(self):
+        """Test for various destination scenarios"""
+
+        filename = 'download_test.txt'
+        test_url = urljoin(self.wdir, 'download_test.txt')
+
+        #destination is directory
+        scraper = mozdownload.DirectScraper(url=test_url,
+                                            destination=self.temp_dir,
```

Comment:
Please add another test which has a non-existent folder as destination, so we can test that the path is correctly created.